src/components/Wallet.tsx

```diff
@@ -150,7 +150,12 @@ const Wallet: React.FC<RouteComponentProps> = (props: WalletProps) => {
       toast.error('Error disconnecting wallet')
     } finally {
       setLoadingLabel(undefined)
-      props.history.push('/')
+
+      // If on trade page already, keep pre-selected values when logging out
+      const pathname = props.location.pathname
+      const redirectTo = typeof pathname === 'string' && pathname.match(/^\/trade/) ? pathname : '/'
+
+      props.history.push(redirectTo)
```

Comment:
Are you still pushing even if on the same path? Why?

Reply:
Also there's a recommended way to check if you are where you think you are location-wise
https://reacttraining.com/react-router/web/api/Hooks/useroutematch

Reply:
Not entirely sure I understood how to use this hook. Can you tell me if that's correct?